azurerm/resource_arm_application_gateway.go

```diff
@@ -2282,6 +2306,40 @@ func flattenApplicationGatewayRedirectConfigurations(input *[]network.Applicatio
 	return results, nil
 }
 
+func expandAutoscaleConfiguration(d *schema.ResourceData) *network.ApplicationGatewayAutoscaleConfiguration {
+	vs := d.Get("autoscale_configuration").([]interface{})
+	if len(vs) == 0 {
+		return nil
+	}
+	v := vs[0].(map[string]interface{})
+
+	minCapacity := int32(v["min_capacity"].(int))
+	maxCapacity := int32(v["max_capacity"].(int))
+
+	configuration := network.ApplicationGatewayAutoscaleConfiguration{
+		MinCapacity: utils.Int32(minCapacity),
+	}
+
+	if maxCapacity != 0 {
+		configuration.MaxCapacity = utils.Int32(maxCapacity)
+	}
+
+	return &configuration
+}
+
+func flattenAutoscaleConfiguration(input *network.ApplicationGatewayAutoscaleConfiguration) []interface{} {
+	result := make(map[string]interface{})
+
```

Comment:
```
------- Stdout: -------
=== RUN   TestAccAzureRMApplicationGateway_zones
=== PAUSE TestAccAzureRMApplicationGateway_zones
=== CONT  TestAccAzureRMApplicationGateway_zones

------- Stderr: -------
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2715e1a]

goroutine 419 [running]:
github.com/terraform-providers/terraform-provider-azurerm/azurerm.flattenAutoscaleConfiguration(...)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_application_gateway.go:2333
github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmApplicationGatewayRead(0xc000554770, 0x342c180, 0xc001008000, 0xc00036cc20, 0x0)
```

Another panic cropped up.

```suggestion
    if input == nil {
        return []interface{}{result}
    }
```